### ISSUE TITLE
Latest Eclipse Oxygen failing with NoSuchFieldError: MODULE (Java 8)

### ISSUE THREAD

**Issue by davido** [google/auto#539]
This was already reported on https://github.com/google/dagger/issues/898. Reporting it here as well.

With the Eclipse Oxygen.1a (4.7.1a), using auto-value with Java 8 (Gerrit Code Review doesn't work with Java 8 yet) building in Eclipse consistently failing for me with this stack trace: [1].

All auto-value versions, started from 1.5, are failing with the same error. The only version that still works with Java 8 is 1.4.1.

To reproduce, clone [gerrit code review](https://gerrit.googlesource.com/gerrit) and follow the documentation how to build it with Bazel and start using it with Eclpse Oxygen 4.7.1a, with Java 8.

* [1] http://paste.openstack.org/show/624262

**Comment by dpursehouse** [comment]
> Gerrit Code Review doesn't work with Java 8 yet

s/Java 8/Java 9/

> clone gerrit code review

...and reset to `88ce029` which is [this change](https://gerrit-review.googlesource.com/c/gerrit/+/133433) that we are in the process of reverting.  It may be that the revert goes in soon, so building from the head of master will no longer fail.

**Comment by paladox** [comment]
I think this is caused by https://github.com/google/auto/commit/2c035ce5172a6c4d21d859e2609b57358db02f6f

Traceback leads me to there from here http://paste.openstack.org/show/624262/

shoulden't we do

import static com.google.auto.common.Visibility.MODULE;?

https://stackoverflow.com/questions/443980/why-cant-enums-constructor-access-static-fields

**Comment by eamonnmcmanus** [comment]
@paladox, the stackoverflow.com question refers to static fields that are referenced from within the enum constructor. The static field in `Visibility` is referenced from with the static method `ofElement` so it should be fine.

This is a [bug](https://bugs.eclipse.org/bugs/show_bug.cgi?id=526288) in Eclipse which you can avoid either by running Eclipse with Java 9 or by using an IDE other than Eclipse.

**Comment by davido** [comment]
The Java 9 APT models was implemented in this CL: [1], in context of this FR: [2]. In the description ist is explicitly stated that:

```
The challenge (again) is going to be, how we will be able to reference the new API without bumping up the project compliance to 9.
```

* [1] https://git.eclipse.org/r/#/c/105339
* [2] https://bugs.eclipse.org/bugs/show_bug.cgi?id=517841

**Comment by davido** [comment]
This was fixed in Eclipse 4.7.2 with: http://git.eclipse.org/c/jdt/eclipse.jdt.core.git/commit/?id=0636dfcefd86dbc60e1602c4836ecf7111822e9c.
